samples/sample_common/src/sample_utils.cpp

```diff
@@ -774,175 +774,191 @@ void CSmplYUVWriter::Close()
-            {
-                    MSDK_CHECK_NOT_EQUAL(
-                        fwrite(pData.U + (pInfo.CropY * pData.Pitch / 2 + pInfo.CropX / 2)+ i * pData.Pitch / 2, 1, pInfo.CropW/2, dstFile),
-                        (mfxU32)pInfo.CropW/2, MFX_ERR_UNDEFINED_BEHAVIOR);
-            }
-            break;
-        }
-        case MFX_FOURCC_NV12:
-        {
-            for (i = 0; i < (mfxU32) pInfo.CropH/2; i++)
-            {
-                    MSDK_CHECK_NOT_EQUAL(
-                        fwrite(pData.UV + (pInfo.CropY * pData.Pitch / 2 + pInfo.CropX / 2) + i * pData.Pitch, 1, pInfo.CropW, dstFile),
-                        pInfo.CropW, MFX_ERR_UNDEFINED_BEHAVIOR);
-            }
-            break;
-        }
-        case MFX_FOURCC_P010:
-        case MFX_FOURCC_P210:
-        {
-            mfxU32 height = pInfo.FourCC == MFX_FOURCC_P010 ? (mfxU32)pInfo.CropH / 2 : (mfxU32)pInfo.CropH;
-
-            for (i = 0; i < height; i++)
-            {
-                mfxU16* shortPtr = (mfxU16*)(pData.UV + (pInfo.CropY * pData.Pitch / 2 + pInfo.CropX) + i * pData.Pitch);
-                if (pInfo.Shift)
-                {
-                    // Convert MS-P010 to P010 and write
-                    tmp.resize(pData.Pitch);
-
-                    for (int idx = 0; idx < pInfo.CropW; idx++)
-                    {
-                        tmp[idx] = shortPtr[idx] >> 6;
-                    }
-
-                    MSDK_CHECK_NOT_EQUAL(
-                        fwrite(&tmp[0], 1, (mfxU32)pInfo.CropW * 2, dstFile),
-                        (mfxU32)pInfo.CropW * 2, MFX_ERR_UNDEFINED_BEHAVIOR);
-
-                }
-                else
-                {
-                    MSDK_CHECK_NOT_EQUAL(
-                        fwrite(shortPtr, 1, (mfxU32)pInfo.CropW * 2, dstFile),
-                        (mfxU32)pInfo.CropW * 2, MFX_ERR_UNDEFINED_BEHAVIOR);
-                }
-            }
-            break;
-        }
-
-        case MFX_FOURCC_RGB4:
-        case 100: //DXGI_FORMAT_AYUV
-        case MFX_FOURCC_A2RGB10:
-        {
-            mfxU8* ptr;
-
-            if (pInfo.CropH > 0 && pInfo.CropW > 0)
-            {
-                w = pInfo.CropW;
-                h = pInfo.CropH;
-            }
-            else
-            {
-                w = pInfo.Width;
-                h = pInfo.Height;
-            }
-
-            ptr = MSDK_MIN( MSDK_MIN(pData.R, pData.G), pData.B);
-            ptr = ptr + pInfo.CropX + pInfo.CropY * pData.Pitch;
-
-            for(i = 0; i < h; i++)
-            {
-                MSDK_CHECK_NOT_EQUAL(fwrite(ptr + i * pData.Pitch, 1, 4*w, dstFile), 4*w, MFX_ERR_UNDEFINED_BEHAVIOR);
-            }
-            fflush(dstFile);
-            break;
-        }
-
-        default:
-            return MFX_ERR_UNSUPPORTED;
-    }
-
-    return MFX_ERR_NONE;
+	MSDK_CHECK_ERROR(m_bInited, false, MFX_ERR_NOT_INITIALIZED);
+	MSDK_CHECK_POINTER(pSurface, MFX_ERR_NULL_PTR);
+
+	mfxFrameInfo &pInfo = pSurface->Info;
```

Comment:
These lines looks to be identical. Are line feeds the same? Are there tabs/space difference?